docs/api_reference/src/quantities.tex

```diff
@@ -5196,18 +5261,18 @@
 \remarks
 The expression in the \fakegrammarterm{requires-clause} is equivalent to:
 \begin{codeblock}
-(Q::unit == ::mp_units::one) && @\placeholder{C}@<Rep, Value>
+(Q::unit == mp_units::one) && @\placeholder{C}@<Rep, Value>
```

Comment:
Is there a good reason for this change? Should I also update the sources? I used `::` to ensure that we will not have `mp_units::mp_units` by accident :wink:

Reply:
> Is there a good reason for this change? Should I also update the sources?

I thought that's how the C++ Standard library did it.
Now I searched for `::std::`, and found that this is also used in 2 places:
- `d.operator ::std::basic_string_view` in `basic_string_view::basic_string_view`.
- In [locale.facet], a requirement on an `id` member.

Besides CPO(-like)s, `move`, `forward`, compatibility subclauses, and examples,
just `std::` is used for
- [format]'s types' use of `std::locale` (since they have a `locale` member).
- `prm`'s `using X = std::X`.
- container(-like)'s `using reverse_iterator = std::reverse_iterator`,
- `span`'s `using const_iterator = std::const_iterator`,
- `span`'s use of `std::data` (since it has a `data` member),
- view's use of `std::min` (since `std::ranges` has a `min`),
- view's use of `std::get` (since `std::ranges` has a `get`),
- [alg.reverse]'s use of `std::iter_swap` (along with `ranges::iter_swap`),
- `linalg` types' use of `std::extents` (since they have an `extents` member),
- I/O types' use of `std::span` (since they have a `span` member),
- `path`'s `std::Xstring Xstring()` members, and
- [future.state]'s use of `std::async` (for no apparent reason),
- [future.async]'s use of `std::bad_alloc` (for no apparent reason),

`std::chrono::zoned_traits`, which has a `locate_zone` member, uses
`std::chrono::locate_zone` instead of
`::std::chrono::locate_zone` or
`chrono::locate_zone`.

So I might prepend `::` to all occurrences of `mp_units::`.
I don't know if it'd be worth it, considering that these names might not live in `::std`.

> I used `::` to ensure that we will not have `mp_units::mp_units` by accident 😉

How is that possible?

Reply:
The probability is low, but there is always a chance to have incorrect curly braces in headers or a user that will do some strange thing. Anyway, I am fine with `mp_units::` as well. I just wanted to check if there is a bigger reason that I am not aware of.

Reply:
That's just bad practice.
Everyone should be following [\[using.headers\]p3](https://eel.is/c++draft/using.headers#3) unless otherwise documented for the header.

I'm making this change right now.
You don't follow it in the specializations for `std::common_type` of `quantity`.